src/main/groovy/org/transmartproject/batch/startup/JobDetails.groovy

```diff
@@ -0,0 +1,284 @@
+            'expression'             : MrnaDataJobSpecification,
+            'metabolomics_annotation': MetabolomicsAnnotationJobSpecification,
+            'metabolomics'           : MetabolomicsDataJobSpecification,
+            'i2b2'                   : I2b2JobSpecification,
+            'proteomics_annotation'  : ProteomicsAnnotationJobSpecification,
+            'proteomics'             : ProteomicsDataJobSpecification,
+            'gwas'                   : GwasJobSpecification,
+            'backout'                : BackoutJobSpecification,
+            'rnaseq_annotation'      : RnaSeqAnnotationJobSpecification,
+            'rnaseq'                 : RnaSeqDataJobSpecification,
+            'acgh_annotation'        : AcghAnnotationJobSpecification,
+            'acgh'                   : AcghDataJobSpecification,
+    ]
+    public static final String STUDY_PARAMS_FILE_NAME = 'study' + PARAMS_FILE_EXTENSION
+    public static final String PARAMS_FILE_EXTENSION = '.params'
+
+    private final Map<String, String> params = Maps.newHashMap()
+
+    private Path dataTypeParamsFilePath
+
+    private String typeName
+
+    private List<? extends ExternalJobParametersModule> modules
+
+    private Set<String> fileParameterKeys
+
+    Class<?> jobPath
+
+    private JobDetails() {}
+
+    /**
+     * Detects the job to run by '*.params' file.
+     * @param filePath - a *.params file
+     * @param overrides - parameters to override ones in the *.params file
+     * @return The job details.
+     */
+    static JobDetails fromFile(Path filePath, Map<String, String> overrides = [:]) {
+
+        String dataType = getDataTypeByProperyFile(filePath)
+        JobSpecification spec = getJobSpecificationByDataType(dataType)
+
+        JobDetails instance = new JobDetails()
+        instance.dataTypeParamsFilePath = filePath
+        instance.typeName = dataType
+        instance.modules = spec.jobParametersModules
+        instance.fileParameterKeys = getFileParameterKeys(spec)
+        instance.jobPath = spec.jobPath
+
+        Map<String, String> dataTypeParams = parseContent(filePath)
+        if (isStudyRelatedSpecification(spec)) {
+            Path studyParamsFilePath = findStudyFilePath(filePath)
+            if (studyParamsFilePath) {
+                Map<String, String> studyParams = parseContent(studyParamsFilePath)
+                dataTypeParams.each { item ->
+                    if (studyParams.containsKey(item.key)) {
+                        log.warn("Study-wise ${item.key} parameter gets overriden by data type one:" +
+                                " ${studyParams[item.value]} -> ${item.value}")
+                    }
+                }
+                instance.params << studyParams
+            } else {
+                log.warn('No study parameters file is found.' +
+                        ' Rely on data type parameters file for specifying study parameters.')
```

Comment:
I don't think this should trigger a warning. I see no reason you should have two params files if you only have clinical data. Plus, I think this will trigger warnings for all test cases.